src/layers/core/choropleth-layer/choropleth-layer.js

```diff
@@ -210,3 +207,89 @@ export default class ChoroplethLayer extends BaseLayer {
   }
 
 }
+
+// convert geojson to choropleths
+// @param {object} data - geojson object
+// @returns {array.array.array.point} array of geojson polygon-like coordinates
+function extractChoropleths(data) {
+  const normalizedGeojson = normalize(data);
+  const result = [];
+
+  normalizedGeojson.features.map((feature, featureIndex) => {
+    const choropleths = featureToChoropleths(feature);
+    choropleths.forEach(choropleth => {
+      choropleth.featureIndex = featureIndex;
+    });
+    result.push(...choropleths);
+  });
+  return result;
+}
+
+// normalize the coordinates array of a geojson feature
+// @param {object} feature - geojson feature object
+// @returns {array.array.array.point} array of geojson polygon-like coordinates
+function featureToChoropleths(feature) {
+  const {coordinates, type} = feature.geometry;
+  let choropleths;
+
+  switch(type) {
+  case 'MultiPolygon':
+    choropleths = coordinates;
+    break;
+  case 'Polygon':
+    choropleths = [coordinates];
+    break;
+  case 'LineString':
```

Comment:
> Would love a robust and generic GeoJSON layer that can render anything in the spec - basically trying to hack one together right now, can contribute code from that if I can split it out

@contra - One problem I see is that a GeoJSON layer might need three separate primitives or models (GL.TRIANGLES for the filled polygon areas, GL.LINES for polygon outlines and LineStrings and and perhaps a GL.POINTS primitive for the Points.)

This could either be handled by having a Layer that uses multiple primitives (on the roadmap but not yet supported by deck.gl) or allow composite layers (also on the roadmap but not yet supported). 

An alternative is to "extrude" all one and zero dimensional objects (i.e. lines and points) into triangles. This would only require one primitive and generate the best visuals. There is an example of how to do this in the dev branch (layers/experimental-layers/extruded-choropleth-layer) but the extrusion used there doesn't currently handle steep angles gracefully.

You are welcome to put up a PR if you make some progress so that we can compare our approaches.